pipeline-model-definition/src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/ModelInterpreter.groovy

```diff
@@ -238,15 +244,16 @@ class ModelInterpreter implements Serializable {
                       SkippedStageReason skippedReason) {
         return {
             script.stage(thisStage.name) {
+                def flowNodeId = getFlowNodeId()
```

Comment:
Still waiting on the full results, but so far this breaks some tests in `blueocean-pipeline-api-impl` and another ~10 spread across some proprietary plugins I have access to:

* `blueocean-pipeline-api-impl / io.jenkins.blueocean.rest.impl.pipeline.PipelineNodeTest.sequentialParallelStagesWithPost`
* `blueocean-pipeline-api-impl / io.jenkins.blueocean.rest.impl.pipeline.PipelineNodeTest.submitInputPostBlock`
* `blueocean-pipeline-api-impl / io.jenkins.blueocean.rest.impl.pipeline.PipelineNodeTest.declarativeSyntheticSteps`
* `blueocean-pipeline-api-impl / io.jenkins.blueocean.rest.impl.pipeline.PipelineNodeTest.encodedStepDescription`
 * `blueocean-pipeline-api-impl / io.jenkins.blueocean.rest.impl.pipeline.PipelineNodeTest.submitInputPostBlockWithParallelStages`
* `blueocean-pipeline-api-impl / io.jenkins.blueocean.rest.impl.pipeline.PipelineNodeTest.declarativeSyntheticSkippedStage`

Maybe we could move the call to `getFlowNodeId` inside of `skipStage` so that the changes only affects tests that are directly related to skipped stages?

Reply:
[bd14425](https://github.com/jenkinsci/pipeline-model-definition-plugin/pull/700/commits/bd144259a8a92027109360538429b4f9e7813e51) will hopefully help with this, but tests like `PipelineNodeTest.declarativeSyntheticSkippedStag` will probably still need to be adjusted.

Reply:
created a test run in https://github.com/jenkinsci/blueocean-plugin/pull/2555

I tried to run locally but I can't get the plugin to build 😢

Reply:
> I tried to run locally but I can't get the plugin to build due to a like 6 year old npm / node version and unmaintained dependencies 😢

Yep, I ran into the same issue in https://github.com/jenkinsci/blueocean-plugin/pull/2540.


With the latest changes, the total affected tests went from 15 to 6, so that's nice. The only remaining affected test in Blue Ocean appears to be `blueocean-pipeline-api-impl / io.jenkins.blueocean.rest.impl.pipeline.PipelineNodeTest.submitInputPostBlockWithParallelStages`. Here's the output: 

```
org.junit.ComparisonFailure: expected:<[47]> but was:<[50]>
	at org.junit.Assert.assertEquals(Assert.java:117)
	at org.junit.Assert.assertEquals(Assert.java:146)
	at io.jenkins.blueocean.rest.impl.pipeline.PipelineNodeTest.submitInputPostBlockWithParallelStages(PipelineNodeTest.java:2314)
```

Let's try to update your Blue Ocean PR to fix the test. There are other tests that will need to be updated internally in CloudBees, but I will take care of that. I'll work on all of this later today.

Reply:
Thanks, I've pushed a guess in https://github.com/timja/blueocean-plugin/commit/5184941cb10b8ae8b6ed0abd2d95a728c5f81bf7

Reply:
> > I tried to run locally but I can't get the plugin to build due to a like 6 year old npm / node version and unmaintained dependencies 😢
> 
> Yep, I ran into the same issue in [jenkinsci/blueocean-plugin#2540](https://github.com/jenkinsci/blueocean-plugin/pull/2540).

Based on a suggestion by @batmat, I tried building this plugin in the environment created by https://github.com/jenkinsci/blueocean-plugin/blob/1ce0c18265882733f3251dcad0b4af616da5a7c9/Dockerfile.build and it built just fine on my AMD64 system. So while definitely annoying, building this plugin locally is not insurmountable on AMD64. Presumably there is some way to run AMD64 code on ARM64 via emulation/virtualization, where the same `Dockerfile.build` could be used.

Reply:
It doesn’t even work on amd64 macOS for me. Sure you can make it work on docker.

I’ve got a branch locally that has made a fair bit of progress but now need to update the JS builder and figure out who has access to release there

Reply:
I got it working in https://github.com/jenkinsci/blueocean-plugin/pull/2558